docs/api.md

```diff
@@ -1,1880 +1,2210 @@
+```js
+{
+  "code": 400,  // Matches the HTTP status code
+  "errno": 107, // Stable application-level error number
+  "error": "Bad Request", // String description of the error type
+  "message": "Invalid parameter in request body", // Specific error message
+  "info": "https://docs.dev.lcip.og/errors/1234"  // Link to more information
+}
+```
 
-## URL Structure
+Responses for some errors may include additional parameters.
+<!--end-response-format-->
+
+#### Defined errors
+
+The currently-defined values
+for `code` and `errno` are:
+
+* `code: 400, errno: 100`:
+  Incorrect Database Patch Level
+* `code: 400, errno: 101`:
+  Account already exists
+* `code: 400, errno: 102`:
+  Unknown account
+* `code: 400, errno: 103`:
+  Incorrect password
+* `code: 400, errno: 104`:
+  Unverified account
+* `code: 400, errno: 105`:
+  Invalid verification code
+* `code: 400, errno: 106`:
+  Invalid JSON in request body
+* `code: 400, errno: 107`:
+  Invalid parameter in request body
+* `code: 400, errno: 108`:
+  Missing parameter in request body
+* `code: 401, errno: 109`:
+  Invalid request signature
+* `code: 401, errno: 110`:
+  Invalid authentication token in request signature
+* `code: 401, errno: 111`:
+  Invalid timestamp in request signature
+* `code: 411, errno: 112`:
+  Missing content-length header
+* `code: 413, errno: 113`:
+  Request body too large
+* `code: 429, errno: 114`:
+  Client has sent too many requests
+* `code: 401, errno: 115`:
+  Invalid nonce in request signature
+* `code: 410, errno: 116`:
+  This endpoint is no longer supported
+* `code: 400, errno: 120`:
+  Incorrect email case
+* `code: 400, errno: 123`:
+  Unknown device
+* `code: 400, errno: 124`:
+  Session already registered by another device
+* `code: 400, errno: 125`:
+  The request was blocked for security reasons
+* `code: 400, errno: 126`:
+  Account must be reset
+* `code: 400, errno: 127`:
+  Invalid unblock code
+* `code: 400, errno: 129`:
+  Invalid phone number
+* `code: 400, errno: 130`:
+  Invalid region
+* `code: 400, errno: 131`:
+  Invalid message id
+* `code: 500, errno: 132`:
+  Message rejected
+* `code: 400, errno: 133`:
+  Email account sent complaint
+* `code: 400, errno: 134`:
+  Email account hard bounced
+* `code: 400, errno: 135`:
+  Email account soft bounced
+* `code: 400, errno: 136`:
+  Email already exists
+* `code: 400, errno: 137`:
+  Can not delete primary email
+* `code: 400, errno: 138`:
+  Unverified session
+* `code: 400, errno: 139`:
+  Can not add secondary email that is same as your primary
+* `code: 400, errno: 140`:
+  Email already exists
+* `code: 400, errno: 141`:
+  Email already exists
+* `code: 503, errno: 201`:
+  Service unavailable
+* `code: 503, errno: 202`:
+  Feature not enabled
+* `code: 500, errno: 999`:
+  Unspecified error
+
+The following errors
+include additional response properties:
+
+* `errno: 100`: level, levelRequired
+* `errno: 101`: email
+* `errno: 102`: email
+* `errno: 103`: email
+* `errno: 105`
```

Comment:
Yep, this is one of the bits that isn't quite right yet.

This error function takes a generic `details` object, so figuring out what those details might be would require parsing all the source for calls to this error function. Not sure I can be arsed with that, so if some kind of "we're not sure" comment is acceptable I could add that. If not, maybe it's another nail in this script's coffin.